qiskit/cluster.py

```diff
@@ -114,7 +194,7 @@ def hadamard(self, global_qubit_index):
         (processor_index, local_qubit_index) = self._global_to_local_index(global_qubit_index)
         self.processors[processor_index].local_hadamard(local_qubit_index)
 
-    def controlled_phase(self, angle, global_control_qubit_index, global_target_qubit_index):
+    def controlled_phase(self, angle, global_control_qubit_index, global_target_qubit_index,flag):
```

Comment:
If you store the flag (method) as in a class attribute, you can avoid passing it around in all these function calls.

Reply:
definitely, I'll make this change